evidence/pool.go

```diff
@@ -235,49 +468,33 @@ func (evpool *Pool) IsPending(evidence types.Evidence) bool {
 	return ok
 }
 
-// EvidenceFront goes to the first evidence in the clist
-func (evpool *Pool) EvidenceFront() *clist.CElement {
-	return evpool.evidenceList.Front()
-}
-
-// EvidenceWaitChan is a channel that closes once the first evidence in the list is there. i.e Front is not nil
-func (evpool *Pool) EvidenceWaitChan() <-chan struct{} {
-	return evpool.evidenceList.WaitChan()
-}
-
-// SetLogger sets the Logger.
-func (evpool *Pool) SetLogger(l log.Logger) {
-	evpool.logger = l
-}
-
-// State returns the current state of the evpool.
-func (evpool *Pool) State() sm.State {
-	evpool.mtx.Lock()
-	defer evpool.mtx.Unlock()
-	return evpool.state
-}
-
-func (evpool *Pool) addPendingEvidence(evidence types.Evidence) error {
-	evi, err := types.EvidenceToProto(evidence)
+func (evpool *Pool) addPendingEvidence(evInfo *Info) error {
+	evpb, err := evInfo.ToProto()
 	if err != nil {
 		return fmt.Errorf("unable to convert to proto, err: %w", err)
 	}
 
-	evBytes, err := proto.Marshal(evi)
+	evBytes, err := evpb.Marshal()
 	if err != nil {
 		return fmt.Errorf("unable to marshal evidence: %w", err)
 	}
 
-	key := keyPending(evidence)
+	key := keyPending(evInfo.Evidence)
 
-	return evpool.evidenceStore.Set(key, evBytes)
+	err = evpool.evidenceStore.Set(key, evBytes)
+	if err != nil {
+		return fmt.Errorf("unable to persist evidence: %w", err)
+	}
+	atomic.AddUint32(&evpool.evidenceSize, 1)
+	return nil
 }
 
 func (evpool *Pool) removePendingEvidence(evidence types.Evidence) {
 	key := keyPending(evidence)
 	if err := evpool.evidenceStore.Delete(key); err != nil {
 		evpool.logger.Error("Unable to delete pending evidence", "err", err)
 	} else {
+		atomic.AddUint32(&evpool.evidenceSize, ^uint32(0))
```

Comment:
^0 ?

Reply:
https://golang.org/src/sync/atomic/doc.go?s=3850:3905#L88

Reply:
apparently that is how it's supposed to be